### ISSUE TITLE
Two undo commands when there is an initial inlineEdit

### ISSUE THREAD

**Issue by JonFerraiolo** [maqetta/maqetta#320]
@billreed63

If you drag/drop a Button onto canvas, you get the inlineEdit control to type
in a new value. If you close the inlineEdit control such that a completed
Button widget is now on the canvas, and hit Undo, nothing happens. Hit Undo
again and the button disappears. This isn't what a user would expect or want.

Somehow or other we need to combine into a single undo command both the
original placement of the Button onto the canvas and the initial inlineEdit
action.

**Comment by billreed63** [comment]
The reason this happens is that the widget is already created and we are basically doing a modify  of the widget. That is why if the user cancels smart input the widget is still on the canvas.  
So after the user clicks the ok the command stack has a Add command and a Modify command on the stack.

**Comment by JonFerraiolo** [comment]
Yes, I understand that's what is happening. That makes sense to us, the programmers, but to a user, he has created a button, and if he hits undo (once), he expects the button to go away. From a user perspective (who doesn't know about AddCommand and ModifyCommand), this is a bug.

@jhpedemonte

**Comment by jhpedemonte** [comment]
That's interesting.  I wonder if in this particular case (drop new widget which immediately pops up smart input) we need to reach back into the undo stack, pull out the Modify and Add commands and merge them into a single command.

**Comment by JonFerraiolo** [comment]
The reach-back approach seems like the #1 candidate at this point. The other approach that comes to mind would be to delay the completion of the AddCommand until after the SmartInput comes down.

**Comment by jhpedemonte** [comment]
-> M6/Med

**Usability** issue.

**Comment by JonFerraiolo** [comment]
I suggest pushing this to M7. Seems too risky for too little benefit for M6. Wouldn't we need to enhance the command logic to pull this off?

**Comment by jhpedemonte** [comment]
-> M7
